### ISSUE TITLE
[java] UnusedPrivateMethod false positive when outer class has method with same name as in inner class

### ISSUE THREAD

**Issue by sratz** [pmd/pmd#3467]
**Affects PMD Version:** >=6.33.0

**Rule:** [UnusedPrivateMethod](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedprivatemethod)

**Description:** Outer class containing an inner class. Both outer and inner contain a method with the same name. Outer class calls the private inner method -> False positiv ('unused private method') on the inner class method.

Looks like this was introduced by #3157.

**Code Sample demonstrating the issue:**

```java
public class OuterClass {

	private final InnerClass o = new InnerClass();

	public void foo() {
		this.o.foo();
	}

	public void notBar() {
		this.o.bar();
	}

	class InnerClass {

		private void foo() { // same name as method in OuterClass -> false positive
		}

		private void bar() { // different name than a method in OuterClass -> ok
		}
	}
}
```

**Expected outcome:** No PMD messages

PMD reports a violation at line 15, but that's wrong. That's a false positive.

**Running PMD through:** *Maven | [eclipse-pmd](https://github.com/eclipse-pmd/eclipse-pmd/)*

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0
